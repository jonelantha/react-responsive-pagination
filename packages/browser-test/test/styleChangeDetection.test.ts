import { TestHarnessPage } from './test-harness-page';

const testHarness = new TestHarnessPage(page, { throwOnError: true });

beforeAll(async () => {
  await testHarness.goto();

  await page.setViewportSize({ width: 500, height: 700 });

  await testHarness.setField('total', 100);

  await testHarness.setField('current', 50);
});

describe('Pagination style change detection', () => {
  test.each([['inherit'], ['24px'], ['40px'], ['inherit']])(
    'adapts correctly when font-size changed to %p',
    async fontSize => {
      await testHarness.editableStyleBlockLocator().evaluate((styleBlock, style) => {
        styleBlock.innerHTML = style;
      }, `.pagination { font-size: ${fontSize}; }`);

      await testHarness.waitForNextFrame();

      const paginationHtml = await page.$eval('ul.pagination', ul => ul.outerHTML);

      expect(paginationHtml).toMatchSnapshot();
    },
  );

  // TODO - test style change triggered within React events
});
