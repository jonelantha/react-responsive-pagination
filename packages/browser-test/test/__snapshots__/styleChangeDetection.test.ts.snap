// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination style change detection adapts correctly when font-size changed to "24px" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">49</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">50</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">51</a></li><li class="page-item"><a class="page-link" href="#">52</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination style change detection adapts correctly when font-size changed to "40px" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">49</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">50</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">51</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination style change detection adapts correctly when font-size changed to "inherit" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">48</a></li><li class="page-item"><a class="page-link" href="#">49</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">50</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">51</a></li><li class="page-item"><a class="page-link" href="#">52</a></li><li class="page-item"><a class="page-link" href="#">53</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination style change detection adapts correctly when font-size changed to "inherit" 2`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">48</a></li><li class="page-item"><a class="page-link" href="#">49</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">50</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">51</a></li><li class="page-item"><a class="page-link" href="#">52</a></li><li class="page-item"><a class="page-link" href="#">53</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;
